### PR TITLE
Generate Seed Data

### PR DESCRIPTION
**Pull request by raul-gracia**
## Description

Generates the seed data necessary to run the kpi on, and also to have available
data on development.
It seems that the migrations were not run when the
new progress table was generated so it updates the db schema too.

It also generates 2 new factories using factory_bot gem, that will be usefull
for testing purposes later and are used on the seeds.

## Trello Card Link

https://trello.com/c/DZtbvrQw

### CODE REVIEW

File: .solargraph.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+---
```
Comment (padv2010) [inline .solargraph.yml:1]: Why do we need .solargraph.yml?

File: Gemfile [hunk 2]
```diff
@@ -11,6 +11,7 @@ gem "jbuilder"
 gem "pg", "~> 1.1"
 gem "puma", "~> 6.3"
 gem "rails", "~> 7.0.4"
+gem "sassc-rails"
```
Comment (padv2010) [inline Gemfile:14]: Why do we need this gem? It does not look related to this PR.

File: db/seeds.rb [hunk 3]
```diff
@@ -1,8 +1,68 @@
...
-#   Character.create(name: "Luke", movie: movies.first)
+
+Applicant.destroy_all
+ApplicantProgress.destroy_all
+
+100.times do |i|
+  # First 50 applicants are salaried trainees, next 50 are teachers
+  route = i < 50 ? "salaried trainee" : "teacher"
+
+  applicant = FactoryBot.create(:applicant, application_route: route, created_at: rand(30..90).days.ago.to_date)
```
Comment (padv2010) [inline db/seeds.rb:10]: As per this seeds file, it seems that we are generating 50 `trainees` and 50 `teachers`.

The issue that I see with this file, is that it is a bit verbose and is not very reusable. This `seeds` file is creating teachers, trainees and different progresses, but these concepts are not encapsulated to be reusable.

We would benefit from [using traits](https://thoughtbot.github.io/factory_bot/traits/using.html), [lists](https://www.rubydoc.info/gems/factory_bot/FactoryBot%2FSyntax%2FMethods:create_list) and some inheritance with the factories.

We should be able to rewrite this file with the following code:

```ruby
%w(trainee teacher).each |route| do
  FactoryBot.create_list(route, 5, :initial_checks_completed)
  FactoryBot.create_list(route, 5, :home_office_checks_completed_at)
  FactoryBot.create_list(route, 5, :school_checks_completed_at)
  FactoryBot.create_list(route, 5, :school_investigation_required)
  FactoryBot.create_list(route, 5, :visa_investigation_required)
end
# I have not tested this code, syntax might be wrong
```

And reuse all the factories across our RSpec tests.

**Review by padv2010 — COMMENTED** [review]
Good work Raul, a few comments for your consideration:

1. There is a [`wip` commit](c7e79116686e7224db1e21b72d39ad7cc13a67d1) that needs to be squash or fixed up. 
2. The commit a5e578262596603b72a453882ca0404a77b7559d would need to be fixed up with the changes that you are reverting.

File: db/seeds.rb (hunk 3 above)
Comment (raul-gracia) [reply db/seeds.rb]: I've refactor all this into a simplified version using traits

File: config/application.rb [hunk 4]
```diff
@@ -31,6 +31,12 @@ class Application < Rails::Application
     # in config/environments, which are processed later.
     #
     # config.time_zone = "Central Time (US & Canada)"
+    config.time_zone = "London"
+    # The default locale is :en and all translations from config/locales/*.rb,yml are auto loaded.
+    config.i18n.enforce_available_locales = false
+    config.i18n.default_locale = :"en-GB"
+    config.i18n.load_path += Dir[Rails.root.join("config/locales/**/*.yml")]
```
Comment (padv2010) [inline config/application.rb:38]: Translations are currently being loaded. Any reason for adding the locales 
manually to the `load_path` ?

Comment (padv2010) [reply config/application.rb:38]: Can we please remove these four lines if they are not needed for seeding data? I find it a bit confusing merging code that is not directly related, because it is difficult to understand why we are adding it.

File: db/seeds.rb [hunk 5]
```diff
@@ -1,8 +1,14 @@
...
-# This file should contain all the record creation needed to seed the database with its default values.
-# The data can then be loaded with the bin/rails db:seed command (or created alongside the database with db:setup).
-#
-# Examples:
-#
-#   movies = Movie.create([{ name: "Star Wars" }, { name: "Lord of the Rings" }])
-#   Character.create(name: "Luke", movie: movies.first)
+
+Rails.logger.debug("Deleting all applicants and applicant progresses...")
+Applicant.destroy_all
```
Comment (padv2010) [inline db/seeds.rb:4]: @raul-gracia as a precaution, it is probably worth doing:

```ruby
Applicant.destroy_all if Rails.env.development?
```

File: db/seeds.rb [hunk 6]
```diff
@@ -1,8 +1,14 @@
...
+ApplicantProgress.destroy_all
```
Comment (padv2010) [inline db/seeds.rb:5]: We have `dependent: destroy` from Applicant to ApplicantProgress, so we can remove line 5.

**Comment by padv2010** [comment]
👍
